### PR TITLE
mon: make the output of ceph df more reasonable

### PR DESCRIPTION
**Pull request by yangruifeng**
add a column 'RAW USED' for ceph df detail
Signed-off-by: Ruifeng Yang <yangruifeng.09209@h3c.com>

### CODE REVIEW

**Comment by jecluis** [comment]
it would be nice to have a test accompanying this patch.

**Comment by yangruifeng** [comment]
@jecluis I'll have a try, thanks.

**Comment by yangruifeng** [comment]
@jecluis done, but, if the global size is too big, running cost of a long time.

**Comment by jecluis** [comment]
@yangruifeng couldn't you just put a file worth a few bytes instead?

**Comment by yangruifeng** [comment]
@jecluis if the file is too small, %USED is 0.

**Comment by liewegas** [comment]
Showing the *user* bytes stored here is a deliberate choice (as is showing hte "max avail" that indicates how many more user bytes can be stored).  If we want to show the raw storage consumed for the pool we should put that in a different column...

**Comment by yangruifeng** [comment]
@liewegas 
yes, i found that the %USED is not correct when i want to show a pie chart of USED. 
it not notional or raw. 
i think that the %USED should be notional-used/notional-pool-total,
and add a different column %RAW USED to show raw-used/raw-total.
what do you think? thanks.

**Comment by liewegas** [comment]
I think just adding RAW USED for each pool (for ceph df detail) would be 
sufficient.  Your pie chart can add them up and compare to the total raw 
(which is the sum of the device sizes).  Would that work?

**Comment by yangruifeng** [comment]
@liewegas done.

File: src/mon/PGMonitor.cc [hunk 1]
```diff
@@ -1333,7 +1333,7 @@ inline string percentify(const float& a) {
 //void PGMonitor::dump_object_stat_sum(stringstream& ss, Formatter *f,
 void PGMonitor::dump_object_stat_sum(TextTable &tbl, Formatter *f,
                                      object_stat_sum_t &sum, uint64_t avail,
-                                     bool verbose) const
+                                     float raw_used_rate, bool verbose) const
 {
   if (f) {
     f->dump_int("kb_used", SHIFT_ROUND_UP(sum.num_bytes, 10));
```
Comment (liewegas) [inline src/mon/PGMonitor.cc]: it should be included in the json output, too

Comment (yangruifeng) [inline src/mon/PGMonitor.cc]: @liewegas done.

**Comment by gregsfortytwo** [comment]
So I've only seen it once, but in http://gitbuilder.sepia.ceph.com/gitbuilder-ceph-tarball-trusty-amd64-basic/log.cgi?log=e5585bc2c93199552e3263394e7c11d33e8f3c40 the test this adds is one of two causing make check to fail.

>../qa/workunits/cephtool/test.sh:31: expect_false: test 0 '!=' 8192

It looks like it can race if stuff is veeeery slow which might be what's going on and I presume there are other solutions (one I see from looking quickly is checking several subsequent map epochs).

**Comment by gregsfortytwo** [comment]
Although I haven't dug into this so it's vaguely possible the prior rbd test failure somehow mucked things up for this test? Seems unlikely though if I understand what this one is doing.

**Comment by yangruifeng** [comment]
@gregsfortytwo maybe 'sleep 10' is not enough, change to 
```
  #wait for update
  cal_raw_used_size=`ceph df detail | grep cephdf_for_test | awk -F ' ' '{printf "%d\n", 2 * $4}'`
  while [ $cal_raw_used_size -eq 0 ]
  do
    cal_raw_used_size=`ceph df detail | grep cephdf_for_test | awk -F ' ' '{printf "%d\n", 2 * $4}'`
  done
```

**Pull request merged**